addons/suggest_locations/models/stock_move_line.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+from collections import defaultdict
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+from .suggest_locations_policy import SUGGEST_LOCATION_REGISTRY
+
+DROPABLE_STATES = ("assigned", "partially_available")
+CONTRAINTS_REQUIRING_A_CHECK = ("enforce", "enforce_with_empty")
+WITH_EMPTY_LOCATIONS = ("suggest_with_empty", "enforce_with_empty")
+VIEW_SET = set(["view"])
+
+
+class MoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _get_policy_class(self, picking_type, preprocessing=False):
+        """Get the policy for suggesting locations
+        Note: Optionally we could cache the results with @lru_cache to avoid always returning the
+        policy class, but this caused issued with closed cursors. To avoid this,
+        you need to update the environment at the same time you return the cached result.
+        Possibly by having another function that returns the cached class and updated environment
+        by policy.env = self.env
+        """
+        try:
+            policy_type = picking_type.u_suggest_locations_policy
+            policy_class = SUGGEST_LOCATION_REGISTRY[policy_type]
+            return policy_class(self.env, preprocessing)
+        except KeyError:
+            raise ValueError(
+                _(f"Policy with name={picking_type.u_suggest_locations_policy} could not be found")
```

Comment:
I don't think we want the name of the policy to be translated, no need for f-string and use `%s`. Also, you can use `policy_type`.